src/storage/s3/storage.c

```diff
@@ -273,6 +315,99 @@ storageS3RequestAsync(StorageS3 *this, const String *verb, const String *uri, St
+        if (this->keyType == storageS3KeyTypeTemp && (this->credExpirationTime - time(NULL)) < S3_CREDENTIAL_RENEW_SEC)
+        {
+            // Set content-length and host headers
+            HttpHeader *credHeader = httpHeaderNew(NULL);
+            httpHeaderAdd(credHeader, HTTP_HEADER_CONTENT_LENGTH_STR, ZERO_STR);
+            httpHeaderAdd(credHeader, HTTP_HEADER_HOST_STR, this->credHost);
+
+            // If the role was not set explicitly or retrieved previously then retrieve it
+            if (this->credRole == NULL)
+            {
+                // Request the role
+                HttpRequest *request = httpRequestNewP(
+                    this->credHttpClient, HTTP_VERB_GET_STR, STRDEF(S3_CREDENTIAL_URI), .header = credHeader);
+                HttpResponse *response = httpRequestResponse(request, true);
+
+                // Not found likely means no role is associated with this instance
+                if (httpResponseCode(response) == HTTP_RESPONSE_CODE_NOT_FOUND)
+                {
+                    THROW(
+                        ProtocolError,
+                        "role to retrieve temporary credentials not found\n"
+                            "HINT: is a valid IAM role associated with this instance?");
+                }
+                // Else an error that we can't handle
+                else if (!httpResponseCodeOk(response))
+                    httpRequestError(request, response);
+
+                // Get role from the text response
+                MEM_CONTEXT_BEGIN(this->memContext)
+                {
+                    this->credRole = strNewBuf(httpResponseContent(response));
+                }
+                MEM_CONTEXT_END();
+            }
+
+            // Retrieve the temp credentials
+            HttpRequest *request = httpRequestNewP(
+                this->credHttpClient, HTTP_VERB_GET_STR, strNewFmt(S3_CREDENTIAL_URI "/%s", strZ(this->credRole)),
+                .header = credHeader);
+            HttpResponse *response = httpRequestResponse(request, true);
+
+            // Not found likely means the role is not valid
+            if (httpResponseCode(response) == HTTP_RESPONSE_CODE_NOT_FOUND)
+            {
+                THROW_FMT(
+                    ProtocolError,
+                    "role '%s' not found\n"
+                        "HINT: is '%s' a valid IAM role associated with this instance?",
+                    strZ(this->credRole), strZ(this->credRole));
+            }
+            // Else an error that we can't handle
+            else if (!httpResponseCodeOk(response))
+                httpRequestError(request, response);
+
+            // Free old credentials
+            strFree(this->accessKey);
+            strFree(this->secretAccessKey);
+            strFree(this->securityToken);
+
+            // Get credentials from the JSON response
+            KeyValue *credential = jsonToKv(strNewBuf(httpResponseContent(response)));
+            (void)credential;
```

Comment:
Casting to void is often used to suppress warnings for unused variables but I don't see why this is done here in (void)credential; - there are no conditionally compiled code sections with the credential variable so can you explain why this is needed here?

Reply:
Looks I put that in so I could get it to compile before writing the rest of the section. Removed.